stac_fastapi/api/stac_fastapi/api/middleware.py

```diff
@@ -87,31 +89,28 @@ def _get_forwarded_url_parts(self, scope: Scope) -> Tuple[str]:
             else:
                 domain = header_host_parts[0]
                 port = None
-        forwarded = self._get_header_value_by_name(scope, "forwarded")
-        if forwarded is not None:
-            parts = forwarded.split(";")
-            for part in parts:
-                if len(part) > 0 and re.search("=", part):
-                    key, value = part.split("=")
-                    if key == "proto":
-                        proto = value
-                    elif key == "host":
-                        host_parts = value.split(":")
-                        domain = host_parts[0]
-                        try:
-                            port = int(host_parts[1]) if len(host_parts) == 2 else None
-                        except ValueError:
-                            # ignore ports that are not valid integers
-                            pass
+
+        if forwarded := self._get_header_value_by_name(scope, "forwarded"):
+            for proxy in forwarded.split(","):
+                if (proto_expr := re.search(r"proto=(?P<proto>http(s)?)", proxy)) and (
+                    host_expr := re.search(
+                        r"host=(?P<host>[\w.-]+)(:(?P<port>\d{1,5}))?", proxy
+                    )
+                ):
+                    proto = proto_expr.groupdict()["proto"]
+                    domain = host_expr.groupdict()["host"]
+                    port_str = host_expr.groupdict().get("port", None)
+
         else:
             domain = self._get_header_value_by_name(scope, "x-forwarded-host", domain)
             proto = self._get_header_value_by_name(scope, "x-forwarded-proto", proto)
             port_str = self._get_header_value_by_name(scope, "x-forwarded-port", port)
-            try:
-                port = int(port_str) if port_str is not None else None
-            except ValueError:
-                # ignore ports that are not valid integers
-                pass
+
+        try:
+            port = int(port_str) if port_str is not None else port
```

Comment:
Because the regex will only catch `integer` we could have `port_str` to be `None` which is why we need to have `else port` here